tasks/build.py

```diff
@@ -203,35 +204,50 @@ def build_jekyll(ctx, branch, owner, repository, site_prefix,
 
 
 @task
-def download_hugo(ctx, version='0.48'):
+def download_hugo(ctx):
+    HUGO_VERSION_PATH = CLONE_DIR_PATH / HUGO_VERSION
+    if HUGO_VERSION_PATH.is_file():
+        LOGGER.info(f'.hugo-version found')
+        hugo_version = ''
+        with HUGO_VERSION_PATH.open() as hugo_vers_file:
+            hugo_version = hugo_vers_file.readline().strip()
+            # escape-quote the value in case there's anything weird
+            # in the .hugo-version file
+            hugo_version = shlex.quote(hugo_version)
+        if hugo_version:
+            LOGGER.info(f'Using hugo version in .hugo-version: {hugo_version}')
+    else:
+        raise RuntimeError(".hugo-version not found")
     '''
     Downloads the specified version of Hugo
     '''
-    LOGGER.info(f'Downloading hugo version {version}')
-    dl_url = (f'https://github.com/gohugoio/hugo/releases/download/'
-              f'v{version}/hugo_{version}_Linux-64bit.tar.gz')
-    response = requests.get(dl_url)
+    LOGGER.info(f'Downloading hugo version {hugo_version}')
+    try:
+        dl_url = (f'https://github.com/gohugoio/hugo/releases/download/'
+                  f'v{hugo_version}/hugo_{hugo_version}_Linux-64bit.tar.gz')
+        response = requests.get(dl_url)
```

Comment:
requests.get() will validate the certificate, but by default will trust all of the root certificates available to it.  This is desirable behavior for a web browser, because who can anticipate all of the sites a person might want to browse, but is an unnecessary attack surface for an application with a known list of external dependencies.  Better to pass in a `verify=` argument that points to a file that contains only the root certificate necessary to validate the certificate used by github.com.